packages/cli/src/tokens/build.ts

```diff
@@ -6,125 +6,99 @@ import chalk from 'chalk';
 
-import * as configs from './build/configs.js';
+import { getConfigsForThemeDimensions } from './build/configs.js';
+import type { BuildConfig, ThemePermutation } from './build/types.js';
 import { makeEntryFile } from './build/utils/entryfile.js';
 
-const { permutateThemes, getConfigs } = configs;
-
 type Options = {
-  /** Design tokens path  */
+  /** Design tokens path */
   tokens: string;
-  /** Output directoru for built tokens */
+  /** Output directory for built tokens */
   out: string;
   /** Generate preview tokens */
   preview: boolean;
   /** Enable verbose output */
   verbose: boolean;
 };
 
-// type FormattedCSSPlatform = { css: { output: string; destination: string }[] };
+export let buildOptions: Options | undefined;
 
 const sd = new StyleDictionary();
 
+/*
+ * Declarative configuration of the build output
+ */
+const buildConfigs = {
+  typography: { sdConfig: 'typographyVariables', dimensions: ['typography'] },
+  'color-mode': { sdConfig: 'colorModeVariables', dimensions: ['mode'] },
+  semantic: { sdConfig: 'semanticVariables', dimensions: ['semantic'] },
+  storefront: {
+    name: 'Storefront preview tokens',
+    sdConfig: 'typescriptTokens',
+    dimensions: ['mode'],
+    options: { outPath: path.resolve('../../apps/storefront/tokens') },
+  },
+  entryFiles: {
+    name: 'CSS entry files',
+    sdConfig: 'semanticVariables',
+    dimensions: ['semantic'],
+    build: async (sdConfigs, { outPath }) => {
+      await Promise.all(
+        sdConfigs.map(async ({ theme }) => {
+          console.log(`👷 ${theme}.css`);
+
+          return makeEntryFile({ theme, outPath, buildPath: path.resolve(`${outPath}/${theme}`) });
+        }),
+      );
+    },
+  },
+} satisfies Record<string, BuildConfig>;
+
 export async function buildTokens(options: Options): Promise<void> {
-  const verbosity = options.verbose ? 'verbose' : 'silent';
+  buildOptions = options;
   const tokensDir = options.tokens;
-  const storefrontOutDir = path.resolve('../../apps/storefront/tokens');
   const outPath = path.resolve(options.out);
 
+  /*
+   * Build the themes
+   */
   const $themes = JSON.parse(fs.readFileSync(path.resolve(`${tokensDir}/$themes.json`), 'utf-8')) as ThemeObject[];
 
-  const relevant$themes = $themes.filter((theme) => {
-    const group = R.toLower(R.defaultTo('')(theme.group));
-    if (group === 'size' && theme.name.toLowerCase() !== 'default') return false;
-
-    return true;
-  });
-
-  const themes = permutateThemes(relevant$themes);
-
-  const typographyThemes = R.filter((val) => val.mode === 'light', themes);
-  const colormodeThemes = R.filter((val) => val.typography === 'primary', themes);
-  const semanticThemes = R.filter((val) => val.mode === 'light' && val.typography === 'primary', themes);
+  // We only use the 'default' theme for the 'size' group
+  const relevant$themes = $themes.filter((theme) =>
+    R.not(theme.group === 'size' && theme.name.toLowerCase() !== 'default'),
```

Comment:
Fix for potential pitfall filtering themeobjects from `$themes`
```suggestion
    R.not(theme.group.toLowerCase() === 'size' && theme.name.toLowerCase() !== 'default'),
```

Reply:
But these names are not normalised in this filter since its dones before `processTheme`. Its done straight on the raw ThemeObjects from `$themes.json` 

Looking at it now this filter does not work for our own `design-tokens` as group is uppercase `Size`. 

Removing it does not alter the build output either so suggest just removing as it seems redundant now?

Reply:
This is true! I guess I didn't complete my thought process, I was planning to move `processTheme` earlier in the "pipeline".

It works right now because the permutation with `size: "default"` comes before `size: "compact"`, due to the order they are defined in $themes.json. Since none of the build configurations depend on the `size` dimension, the first one is always used.

Reply:
Retaining the `_` I guess was to keep the same name in Figma and denote that they are on the same level. While `-` in our other css-vars denoted next level/group. 

<img width="166" alt="image" src="https://github.com/user-attachments/assets/1cc83199-ac72-47d7-85ef-ae5bfade49e1">


I dont have a strong preference here, so I suggest we suggest renaming this to `validation-message` in `design-tokens` when we fix the names. We have to do fix the name in a separate PR as we are going to touch some other stuff aswell then.

Reply:
But aye, support moving the filter to `processTheme`